docs/src/tutorials/temporal-properties.md

```diff
@@ -0,0 +1,416 @@
+State 3: state invariant 0 violated.                              E@18:04:16.346
+Found 1 error(s)                                                  I@18:04:16.347
+The outcome is: Error                                             I@18:04:16.353
+Checker has found an error                                        I@18:04:16.354
+It took me 0 days  0 hours  0 min  2 sec                          I@18:04:16.354
+Total time: 2.542 sec                                             I@18:04:16.354
+```
+
+This time, we get a counterexample.
+Let's take a look at `/home/user/apalache/docs/src/tutorials/_apalache-out/TrafficLight.tla/2022-05-30T18-04-13_3349613574715319837/counterexample1.tla`.
+
+Let's first focus on the initial state.
+
+```
+(* Initial state *)
+(* State0 ==
+  RequestWillBeFulfilled_init = FALSE
+    /\ __loop_InLoop = FALSE
+    /\ __loop_☐(requestedGreen ⇒ ♢isGreen) = FALSE
+    /\ __loop_requestedGreen ⇒ ♢isGreen = TRUE
+    /\ __loop_♢isGreen = FALSE
+    /\ __loop_isGreen = FALSE
+    /\ __loop_requestedGreen = FALSE
+    /\ ☐(requestedGreen ⇒ ♢isGreen) = FALSE
+    /\ ☐(requestedGreen ⇒ ♢isGreen)_unroll = TRUE
+    /\ requestedGreen ⇒ ♢isGreen = TRUE
+    /\ ♢isGreen = FALSE
+    /\ ♢isGreen_unroll = FALSE
+    /\ isGreen = FALSE
+    /\ requestedGreen = FALSE *)
+State0 ==
+  RequestWillBeFulfilled_init = FALSE
+    /\ __loop_InLoop = FALSE
+    /\ __loop___temporal_t_1 = FALSE
+    /\ __loop___temporal_t_2 = TRUE
+    /\ __loop___temporal_t_3 = FALSE
+    /\ __loop_isGreen = FALSE
+    /\ __loop_requestedGreen = FALSE
+    /\ __temporal_t_1 = FALSE
+    /\ __temporal_t_1_unroll = TRUE
+    /\ __temporal_t_2 = TRUE
+    /\ __temporal_t_3 = FALSE
+    /\ __temporal_t_3_unroll = FALSE
+    /\ isGreen = FALSE
+    /\ requestedGreen = FALSE
+```
+
+Two things are notable:
+1. The initial state formula appears twice, once as a comment and once in TLA+.
+2. There are way more variables than the two variables we specified.
+
+The comment and the TLA+ specification express the same state, but in the comment, some variable names from the encoding have been replaced
+with more human-readable names.
+For example, there is a variable called `☐(requestedGreen ⇒ ♢isGreen)` in the comment,
+which is called `__temporal_t_1` in TLA+.
+In the following, let's focus on the content of the comment, since it's easier to understand what's going on.
+
+There are many additional variables in the counterexample because to check temporal formulas, Apalache uses an
+encoding that transforms temporal properties to invariants.
+If you are interested in the technical details, the encoding is described in sections 3.2 and 4 of [Biere et al.][].
+However, to understand the counterexample, you don't need to go into the technical details of the encoding.
+We'll go explain the counterexample in the following.
+
+## Counterexamples encode traces with a loop
```

Comment:
The notion of a trace appears out of nowhere. We usually use the word "computation" and the canonical TLA+ notion is that of a behavior. I think you are right in that we have to talk about traces here, as we decorate computation states with additional predicates. It would be good to have the notion of a trace quickly introduced in one paragraph.

Reply:
The nomenclature is from https://apalache.informal.systems/docs//tutorials/symbmc.html?highlight=trace#symbolic-traces, so I'll just link that